File: Python3/1016_binary_string_with_substrings_representing_1_to_n.py
# https://leetcode.cn/problems/binary-string-with-substrings-representing-1-to-n/

class Solution:
    def queryString(self, s, n):
        """
        Inputs:
            s: str
            n: int
        Outputs:
            res: bool
        """
        check = set()
        for i in range(len(s)):
            num = int(s[i])
            if num <= n and num > 0:
                check.add(num)
            else:
                continue
            for j in range(i+1, len(s)):
                num = num * 2 + int(s[j])
                if num <= n and num > 0:
                    check.add(num)
                else:
                    break
        return len(check) == n

File: Cpp/1016_binary_string_with_substrings_representing_1_to_n.cpp
// https://leetcode.cn/problems/binary-string-with-substrings-representing-1-to-n/

class Solution {
public:
    bool queryString(string s, int n) {
        unordered_set<int> check;
        for (int i = 0; i < s.size(); ++i) {
            int num = s[i] - '0';
            if (num <= n && num > 0) check.insert(num);
            else continue;
            for (int j = i+1; j < s.size(); ++j) {
                num = num * 2 + s[j] - '0';
                if (num <= n && num > 0) check.insert(num);
                else break;
            }
        }
        return check.size() == n? true: false;
    }
};
